mountpoint-s3/src/cli.rs

```diff
@@ -632,10 +652,13 @@ where
     filesystem_config.storage_class = args.storage_class;
     filesystem_config.allow_delete = args.allow_delete;
     filesystem_config.allow_overwrite = args.allow_overwrite;
-    filesystem_config.s3_personality = s3_personality;
-    {
-        filesystem_config.server_side_encryption = ServerSideEncryption::new(args.sse, args.sse_kms_key_id);
+    filesystem_config.use_upload_checksums = !args.disable_upload_checksums;
+    if !s3_personality.supports_additional_checksums() {
+        tracing::info!("disabling upload checksums because target S3 personality does not support them");
+        filesystem_config.use_upload_checksums = false;
```

Comment:
It would be nice if customers didn't have to upgrade Mountpoint when Outpost starts supporting it. Or you think we'd need more changes in MP to support it anyways?

Reply:
I don't expect we'd need more changes, but I think it's better for us to do this detection now and remove it in a future upgrade when ready. Otherwise, Outposts customers would have to add the disable flag themselves today, and then remember to remove it in the future. This way, we can do that upgrade for them automatically when ready.

Reply:
We now have a way to let customers opt-in to checksums on older versions too now, nice. Resolving this one.